workspace_visualization/src/reachability_map_visual.cpp

```diff
@@ -0,0 +1,215 @@
+#include <OGRE/OgreVector3.h>
+#include <OGRE/OgreSceneNode.h>
+#include <OGRE/OgreSceneManager.h>
+
+#include <rviz/ogre_helpers/arrow.h>
+#include <rviz/ogre_helpers/shape.h>
+#include <rviz/display_context.h>
+#include <rviz/display_factory.h>
+
+#include<tf2/LinearMath/Quaternion.h>
+
+#include "reachability_map_visual.h"
+#include <iterator>
+
+namespace workspace_visualization
+{
+
+
+ReachMapVisual::ReachMapVisual( Ogre::SceneManager* scene_manager, Ogre::SceneNode* parent_node, rviz::DisplayContext* display_context)
+{
+  scene_manager_ = scene_manager;
+  frame_node_ = parent_node->createChildSceneNode();
+  
+  //arrow_.reset(new rviz::Arrow( scene_manager_, frame_node_ ));
+}
+
+ReachMapVisual::~ReachMapVisual()
+{
+  scene_manager_->destroySceneNode( frame_node_ );
+}
+
+void ReachMapVisual::setMessage( const map_creator::WorkSpace::ConstPtr& msg, bool do_display_arrow, bool do_display_sphere, int low_ri, int high_ri, int shape_choice, int disect_choice)
+{
+ int low_SphereSize, up_SphereSize;
+  if(disect_choice == 0)
+    {
+	low_SphereSize = 0;
+        up_SphereSize = msg->WsSpheres.size();
+    }
+  if(disect_choice == 1)
+    { 
+        low_SphereSize = 0;
+        up_SphereSize = msg->WsSpheres.size()/2;
+    }
+  if(disect_choice == 2)
+    { 
+        low_SphereSize = msg->WsSpheres.size()/2;
+        up_SphereSize = msg->WsSpheres.size();
+    }
+  if(disect_choice == 3)
+    { 
+        low_SphereSize = msg->WsSpheres.size()/2.2;
+        up_SphereSize = msg->WsSpheres.size()/1.8;
+    }
+  if(disect_choice == 4)
```

Comment:
Could use a switch statement here.  Also, how is the dissect choice made?  rosparam? Interactive marker?  Would it be possible to provide a range and allow the user to select an axis and an upper/lower bound to slice with?  If it is possible, create an issue so it can be addressed later.